tests/nested/classic/hotplug/task.yaml

```diff
@@ -0,0 +1,297 @@
+    # Check that given slot has hotplug-gone=true, meaning the device was unplugged but there are connections remembered for it
+    check_slot_gone() {
+        SLOT_NAME="$1"
+        execute_remote 'sudo jq -r ".data[\"hotplug-slots\"][\"'"$SLOT_NAME"'\"][\"hotplug-gone\"]" /var/lib/snapd/state.json' | MATCH "true"
+    }
+
+    # Check that given slot has hotplug-gone=false, meaning the device is plugged
+    check_slot_not_gone() {
+        SLOT_NAME="$1"
+        execute_remote 'sudo jq -r ".data[\"hotplug-slots\"][\"'"$SLOT_NAME"'\"][\"hotplug-gone\"]" /var/lib/snapd/state.json' | MATCH "false"
+    }
+
+    # Check that given slot has no record in "hotplug-slots" map in the state
+    check_slot_not_present_in_state() {
+        SLOT_NAME="$1"
+        execute_remote 'sudo jq -r ".data[\"hotplug-slots\"][\"'"$SLOT_NAME"'\"] // \"missing\"" /var/lib/snapd/state.json' | MATCH "missing"
+    }
+
+    check_slot_device_path() {
+        SLOT_NAME="$1"
+        DEVICE_PATH="$2"
+        execute_remote 'sudo jq -r ".data[\"hotplug-slots\"][\"'"$SLOT_NAME"'\"][\"static-attrs\"].path" /var/lib/snapd/state.json' | MATCH "$DEVICE_PATH"
+    }
+
+    # Check that given slot is connected to the serial-port-hotplug snap, per 'snap connections' output
+    check_slot_connected() {
+        SLOT_NAME="$1"
+        for _ in $(seq 10); do
+            if execute_remote "snap connections" | MATCH "serial-port .*serial-port-hotplug:serial-port .*$SLOT_NAME"; then
+                break
+            fi
+            sleep 1
+        done
+        execute_remote "snap connections" | MATCH "serial-port .*serial-port-hotplug:serial-port .*$SLOT_NAME"
+    }
+
+    # Check that apparmor profile allows rw access to given device path.
+    verify_apparmor_profile() {
+        DEVPATH=$1
+        for _ in $(seq 10); do
+            if execute_remote "cat /var/lib/snapd/apparmor/profiles/snap.serial-port-hotplug.consumer" | MATCH "$DEVPATH rw,"; then
+                break
+            fi
+            sleep 1
+        done
+        execute_remote "cat /var/lib/snapd/apparmor/profiles/snap.serial-port-hotplug.consumer" | MATCH "$DEVPATH rw,"
+    }
+
+    wait_for_all_changes() {
+        for _ in $(seq 10); do
+            if ! execute_remote "snap changes" | MATCH "Doing"; then
+                break
+            fi
+            sleep 1
+        done
+    }
+
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+
+    if execute_remote "udevadm info -e" | MATCH "ID_MODEL.QEMU_USB_SERIAL"; then
+        echo "USB serial already registered, exiting..."
+        exit 1
+    fi
+
+    execute_remote "sudo snap install hello-world"
+
+    echo "Enabling hotplug"
+    execute_remote "sudo snap set core experimental.hotplug=true"
+
+    echo "Plugging the device"
+    hotplug_add_dev1
+
+    # sanity checks to make sure qemu setup is correct
+    for _ in $(seq 5); do
+        if execute_remote "udevadm info -e" | MATCH "ID_MODEL.QEMU_USB_SERIAL"; then
```

Comment:
Same question about the dot above.

Reply:
Yep, good point, should simply use a precise match.